awx/main/tasks/system.py

```diff
@@ -788,10 +797,16 @@ def update_inventory_computed_fields(inventory_id):
     try:
         i.update_computed_fields()
     except DatabaseError as e:
-        if 'did not affect any rows' in str(e):
-            logger.debug('Exiting duplicate update_inventory_computed_fields task.')
-            return
-        raise
+        cause = e.__cause__
+        if cause and hasattr(cause, 'sqlstate'):
+            sqlstate = cause.sqlstate
+            sqlstate_str = psycopg.errors.lookup(sqlstate)
+            logger.error('SQL Error state: {} - {}'.format(sqlstate, sqlstate_str))
+        # To my knowledge no database or library raises an expcetion for "did not affect any rows".
+        # xxx.rowcount on update/delete shoud be used to determined affected rows.
+        # Not sure how an expcetion is thrown from update_computed_fields()
+        else:
+            logger.debug('Exiting duplicate update_inventory_computed_fields task. {}'.format(str(e)))
```

Comment:
You've lost the `raise` in this exception handler.

Reply:
added raise on 805

Reply:
FYI, the "did not affect any rows" message is coming from Django instead of the db in this case: https://github.com/django/django/blob/eff21d8e7a1cb297aedf1c702668b590a1b618f3/django/db/models/base.py#L1107

It's that one in particular, since `i.update_computed_fields()` makes use of `.save(update_fields=...)`.

You are correct that in this case the exception won't get a `sqlstate`, so this flow of logic looks ok to me.